### ISSUE TITLE
Specs2 incompatible with newest version of Mockito

### ISSUE THREAD

**Issue by beaushinkle** [etorreborre/specs2#641]
I'm running into a Mockito issue where mocks of singletons can cause deadlock race conditions. See https://github.com/mockito/mockito/issues/1067

Upgrading mockito to the newly released 2.16.1, however, causes:
```
Exception in thread "specs2-4" java.lang.NoSuchMethodError: org.mockito.internal.invocation.ArgumentsProcessor.expandArgs(Lorg/mockito/internal/invocation/MockitoMethod;[Ljava/lang/Object;)[Ljava/lang/Object;
	at org.mockito.internal.invocation.InterceptedInvocation.<init>(InterceptedInvocation.java:44)
	at org.mockito.internal.invocation.DefaultInvocationFactory.createInvocation(DefaultInvocationFactory.java:42)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:63)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:128)
```

**Comment by alielgamal** [comment]
same issue here.

**Comment by etorreborre** [comment]
I have a fix for mockito `2.16.0` (I could not find `2.16.0`) on master but I cannot publish at the moment because of a setup issue on my new laptop.

**Comment by etorreborre** [comment]
I released `4.0.4-3593406-20180327185327` with the upgrade.

**Comment by etorreborre** [comment]
I just published `4.1.0` with dependency to Mockito `2.18.0`. So this problem should be solved.
